sklearn/model_selection/_search.py

```diff
@@ -344,6 +344,37 @@ def _check_param_grid(param_grid):
                 )
 
 
+def _check_refit(search_cv, attr):
+    if not search_cv.refit:
+        raise AttributeError(
+            f"This {type(search_cv).__name__} instance was initialized with "
+            f"`refit=False`. {attr} is available only after refitting on the best "
+            "parameters. You can refit an estimator manually using the "
+            "`best_params_` attribute"
+        )
+
+
+def _estimator_has(attr):
+    """Check if we can delegate a method to the underlying estimator.
+
+    Calling a prediction method will only be available if `refit=True`. In
+    such case, we check first the fitted best estimator. If it is not
+    fitted, we check if the unfitted estimator.
+    """
+
+    def check(self):
+        _check_refit(self, attr)
+        if hasattr(self, "best_estimator_"):
+            # raise an AttributeError if `attr` does not exist
+            getattr(self.best_estimator_, attr)
+            return True
+        # raise an AttributeError if `attr` does not exist
+        getattr(self.estimator, attr)
```

Comment:
May we include the reason we check for `attr` in `self.estimator`?

I suspect this check exists so that `predict` and friends exist before the `*CV` is fitted?

Reply:
It could also be used by third-party library to check that whatever attribute exists before fit. I think that in meta-estimator it is indeed useful

Reply:
Oh sorry, I see that this is in `_search.py`. In this case, you are right. It is the same spirit if you want to use the object as an estimator.